Owl/app/src/main/java/com/example/owl/ui/NavGraph.kt

```diff
@@ -41,40 +65,126 @@ object MainDestinations {
 }
 
 @Composable
-fun NavGraph(startDestination: String = MainDestinations.ONBOARDING_ROUTE) {
+fun NavGraph(
+    finishActivity: () -> Unit,
+    startDestination: String = MainDestinations.COURSES_ROUTE,
+    showOnboardingInitially: Boolean = true
+) {
     val navController = rememberNavController()
 
+    // Onboarding could be read from shared preferences.
+    val onboardingComplete = remember { mutableStateOf(!showOnboardingInitially) }
```

Comment:
Shouldn't this be `remember(showOnboardingInitially)`?

Reply:
Why? it has to be state in order to be passed to `courses`

Reply:
Yes but shouldn't it be updated if the `showOnboardingInitially` param changes i.e.
```suggestion
    val onboardingComplete = remember(showOnboardingInitially) { mutableStateOf(!showOnboardingInitially) }
```

Reply:
ah, I read it as `remember  { showOnboardingInitially }`. I don't think that param is ever going to change, but sure.